### ISSUE TITLE
More than one condRef in Deseq2

### ISSUE THREAD

**Issue by ankushs0128** [PF2-pasteur-fr/SARTools#92]
Hi , 

I have more than one condRef for pairwise comparison in Deseq2. I am receiving an error 

> invalid :subscript contains invalid names

. Previously I managed to run this analysis with multiple condRef.  

Any pointers on how can I resolve this error for  multiple condition. 

Thanks in advance

**Comment by hvaret** [comment]
Hi,
actually the condRef parameter is designed to have only one reference condition, but then in the analysis SARTools will test for all the pairwise comparisons. My advice is to order your conditions in the target file so that the pairwise comparisons make sense.
Best,
Hugo

**Comment by ankushs0128** [comment]
Hi ,
Metadata looks something like this with many approximate 10 more conditions to compare
<html xmlns:v="urn:schemas-microsoft-com:vml"
xmlns:o="urn:schemas-microsoft-com:office:office"
xmlns:x="urn:schemas-microsoft-com:office:excel"
xmlns="http://www.w3.org/TR/REC-html40">

<head>

<meta name=ProgId content=Excel.Sheet>
<meta name=Generator content="Microsoft Excel 15">
<link id=Main-File rel=Main-File
href="file:////Users/ankushs/Library/Group%20Containers/UBF8T346G9.Office/TemporaryItems/msohtmlclip/clip.htm">
<link rel=File-List
href="file:////Users/ankushs/Library/Group%20Containers/UBF8T346G9.Office/TemporaryItems/msohtmlclip/clip_filelist.xml">



</head>

<body link="#0563C1" vlink="#954F72">



label | Donor | condition
-- | -- | --
Sample1 | patient1 | CD4
Sample2 | patient2 | CD4
Sample3 | patient1 | CD8
Sample4 | patient2 | CD8
Sample5 | patient1 | Treg
Sample6 | patient2 | Treg
Sample5 | patient1 | CD42Treg
Sample6 | patient2 | CD42Treg
Sample5 | patient1 | TregCD82
Sample6 | patient2 | TregCD82



</body>

</html>



```

The parameters used are 
varInt <- "condition"                              # factor of interest
condRef <- "Treg"                                 # reference biological condition
batch <- NULL                                       # blocking factor: NULL (default) or "batch" for example

fitType <- "parametric"                              # mean-variance relationship: "parametric" (default) or "local"
cooksCutoff <- TRUE                                 # TRUE/FALSE to perform the outliers detection (default is TRUE)
independentFiltering <- TRUE                  # TRUE/FALSE to perform independent filtering (default is TRUE)
alpha <- 0.05                                              # threshold of statistical significance
pAdjustMethod <- "BH"                            # p-value adjustment method: "BH" (default) or "BY"

typeTrans <- "rlog"                                   # transformation for PCA/clustering: "VST" or "rlog"
locfunc <- "median"                                 # p-value adjustment method: "BH" (default) or "BY"
    
```                          
The pairwise comparison with reference conditions is computed but pairwise comparison CD4 vsCD8 throws this error.  

`Error: subscript contains invalid names`

Could you please point me how to order these conditions in the metadata file?

**Comment by hvaret** [comment]
Hi,
the error is strange and I'm not sure it comes from your design. If your data are not confidential could you please them by e-mail so that I can reproduce the error message?
Best,
Hugo

**Comment by ankushs0128** [comment]
Hi,  
Sent you an email with an example test dataset.  

Best 
Ankush

**Comment by hvaret** [comment]
Hi,
thanks for your e-mail. Actually there was a bug in SARTools so I corrected it and pushed a new version (1.8.1) on the master branch. Could you please install it and let me know if it works?
Best,
Hugo

**Comment by ankushs0128** [comment]
Thanks, it worked
